tgui/packages/tgui/components/Modal.js

```diff
@@ -6,10 +6,18 @@ export const Modal = props => {
   const {
     className,
     children,
+    onEnter,
     ...rest
   } = props;
+  let handleKeyDown;
+  if (onEnter) {
+    handleKeyDown = e => {
+      onEnter(e);
```

Comment:
Hey, as a heads up, I think you meant to check the keyCode value being equal to 13 here, but didn't. As it is, any key press (including regular typing) that the Modal manages to pick up (which it will from using the input) will trigger onEnter() and instantly submit the input.

Reply:
*screams internally*